mdformat/plugins.py

```diff
@@ -15,3 +20,33 @@ def _load_codeformatters() -> Dict[str, Callable[[str, str], str]]:
 
 
 CODEFORMATTERS: Mapping[str, Callable[[str, str], str]] = _load_codeformatters()
+
+
+class ParsePluginAbstract:
+    """A class for extending the base parser."""
+
+    @staticmethod
+    def update_mdit(mdit: MarkdownIt) -> MarkdownIt:
+        """Update the parser, e.g. by adding a plugin: `mdit.use(myplugin)`"""
+        return mdit
+
+    @staticmethod
+    def render_token(
+        renderer: MDRenderer, tokens: List[Token], index: int, options: dict, env: dict
```

Comment:
Is there a use case for the `renderer` arg here?

Reply:
definitely; see my explanation in https://github.com/executablebooks/mdformat/pull/29#issuecomment-698216949, whereby for the full tables renderer (and likely other extensions) you will need to perform nested rendering.

Reply:
Thanks, got it!